pkgtools/pkg_filecheck/files/builddb.1
.\"	$NetBSD$
.\"
.\" Copyright (c) 2003 Thomas Klausner
.\" All rights reserved.
.\"
.\" Redistribution and use in source and binary forms, with or without
.\" modification, are permitted provided that the following conditions
.\" are met:
.\" 1. Redistributions of source code must retain the above copyright
.\"    notice, this list of conditions and the following disclaimer.
.\" 2. Redistributions in binary form must reproduce the above copyright
.\"    notice, this list of conditions and the following disclaimer in the
.\"    documentation and/or other materials provided with the distribution.
.\"
.\" THIS SOFTWARE IS PROVIDED BY THE AUTHOR, THOMAS KLAUSNER,
.\" ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
.\" TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
.\" PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE FOUNDATION OR CONTRIBUTORS
.\" BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
.\" CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
.\" SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
.\" INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
.\" CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
.\" ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
.\" POSSIBILITY OF SUCH DAMAGE.
.\"
.Dd July 13, 2003
.Dt BUILDDB 1
.Os
.Sh NAME
.Nm builddb
.Nd create database from file system or tar file
.Sh SYNOPSIS
.Nm
.Op Fl hv
.Op Fl o Ar db-file
.Ar tar-file | Ar directory
.Op Ar ...
.Sh DESCRIPTION
.Nm
reads the tar file
.Ar tar-file
or the contents of the directory hierarchy starting at
.Ar directory
and creates a database from it which can be used by
.Nm pkg_filecheck .
.Pp
Supported options:
.Bl -tag -width XoXdbXfileX -offset indent
.It Fl h
Display a short help message.
.It Fl o Ar db-file
Use
.Ar db-file
as database output file instead of the default
.Pa /var/db/pkg/xdb.byfile.db .
.It Fl v
Verbose mode.
Report each file found.
.El
.Sh SEE ALSO
.Xr pkg_filecheck 1
.Sh BUGS
.Nm
is written in Perl.


doc/NEWS
# $NetBSD: NEWS,v 1.1 2008/01/18 05:47:30 tnn Exp $

PKGSRC NEWS
===========

  This file contains information about important infrastructural changes in
  pkgsrc. The intended readers are the pkgsrc developers as well as anyone
  tracking pkgsrc-current.


OpenSSL update [tnn 20080118]
	OpenSSL in pkgsrc switched from the 0.9.7 to the 0.9.8 branch. To
	accomodate for this, 596 packages dependent on openssl had their
	revisions bumped.

pkgsrc-2007Q4 released [agc 20080105]
	A new stable branch was released. Read the announcement at:
	http://mail-index.netbsd.org/netbsd-announce/2008/01/05/0000.html

Simplifying meta-packages [rillig 20071219]
	The variable META_PACKAGE was introduced. This simplifies creation
	and maintenance of packages that do not install files.

Replace a package with different name [gdt 20071107]
	The variable OLDNAME was introduced for "make replace" to
	replace a package with a name other than the new one, via
	"make replace OLDNAME=oldpkgname".  Read the announcement at:
	http://mail-index.netbsd.org/tech-pkg/2007/11/07/0002.html
	
pkgsrc-2007Q3 released [agc 20071015]
	A new stable branch was released. Read the announcement at:
	http://mail-index.netbsd.org/netbsd-announce/2007/10/15/0000.html

Features framework [jlam 20070907]
	A "system features" framework has been added. It will be used to
	automatically supply missing "basic" headers and libraries for old
	systems. This is supported through the USE_FEATURES variable.

X.org 6.9 removed [joerg 20070817]
	monolithic X.org 6.9 was removed from pkgsrc. It was obsoleted by the
	modular X.org (7.x) packages. Transitioning users should remove all old
	X.org packages and set X11_TYPE=modular in mk.conf.

Emulator framework [jlam 20070729]
	A new emulator framework has been added. This will handle all
	binary-only packages that require emulation on the native OS.
	Please see pkgsrc/mk/emulator/README.

audit-packages merged with pkg_install [adrianp 20070714]
	The utility for auditing installed packages for security problems,
	audit-packages, has been rewritten in C. It is now part of the
	pkgtools/pkg_install package.
	
pkgsrc-2007Q2 released [agc 20070701]
	A new stable branch was released. Read the announcement at:
	http://mail-index.netbsd.org/netbsd-announce/2007/07/01/0000.html

New framework for bulk builds [joerg 20070619]
	pbulk, the new automated pkgsrc bulk build framework was imported.

pkgsrc-2007Q1 released [agc 20070419]
	A new stable branch was released. Read the announcement at:
	http://mail-index.netbsd.org/netbsd-announce/2007/04/19/0000.html

pkgsrc-2006Q4 released [agc 20070104]
	A new stable branch was released. Read the announcement at:
	http://mail-index.netbsd.org/netbsd-announce/2007/01/04/0000.html

modular X.org imported [joerg 20061114]
	The modular X.org packages were imported.

pkgsrc-2006Q3 released [agc 20061001]
	A new stable branch was released. Read the announcement at:
	http://mail-index.netbsd.org/netbsd-announce/2006/10/01/0000.html


pkgtools/pkg_filecheck/files/builddb.pl
#!@PERL5@

#  $NetBSD: builddb.pl,v 1.1.1.1 2003/07/13 16:15:34 wiz Exp $
#
#  builddb -- build .byfile.db from tgz files or directory hierarchies.
#  Copyright (C) 2000, 2003 Dieter Baron
#
#  This file is part of img2eps, an image to EPS file converter.
#  The author can be contacted at <dillo@giga.or.at>
#
#  Redistribution and use in source and binary forms, with or without
#  modification, are permitted provided that the following conditions
#  are met:
#  1. Redistributions of source code must retain the above copyright
#     notice, this list of conditions and the following disclaimer.
#  2. Redistributions in binary form must reproduce the above copyright
#     notice, this list of conditions and the following disclaimer in
#     the documentation and/or other materials provided with the
#     distribution.
#  3. The name of the author may not be used to endorse or promote
#     products derived from this software without specific prior
#     written permission.
# 
#  THIS SOFTWARE IS PROVIDED BY DIETER BARON ``AS IS'' AND ANY EXPRESS
#  OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
#
#  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
#  ARE DISCLAIMED.  IN NO EVENT SHALL DIETER BARON BE LIABLE FOR ANY
#  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
#  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
#  GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
#  IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
#  OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
#  IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

use strict;
use Getopt::Std;
use DB_File;

my $prg;
my $options = 'hvo:';
my $usage = '[-hv] [-o db-file] tar-file|directory ...';
my %opt;
my ($verbose, $dbfile);
my %db;
my ($file, $pkg, $opkg);

#
# defaults
#

$verbose = 0;
$dbfile = '/var/db/pkg/xdb.byfile.db';

#
# command line handling
#

($prg) = ($0 =~ /([^\/]+)$/);

getopts($options, \%opt);

if ($opt{h}) {
    print <<EOF;
usage: $prg $usage
  -h           display this help message
  -o DB-FILE   output to DB-FILE
  -v           verbose mode
EOF
    exit 0;
}

if (@ARGV == 0) {
    print STDERR "usage: $prg $usage\n";
    exit 1;
}

$verbose = $opt{v} if ($opt{v});
$dbfile = $opt{o} if ($opt{o});

#
# setup
#

tie %db, 'DB_File', $dbfile, O_RDWR|O_CREAT, 0666, $DB_BTREE
	or die "cannot open db `$dbfile': $!\n";

#
# process tar files
#

foreach $file (@ARGV) {
    if (-f $file) {
	unless (open LST, "tar -tzf $file |") {
	    warn "cannot list `$file': $!\n";
	    next;
	}
	
	($pkg) = ($file =~ m/([^\/]+)\.tgz$/);
    }
    elsif (-d $file) {
	$pkg = $file;

	$file = "`pwd`/$file" unless ($file =~ m!^/!);
	unless (open LST, "find $file -type f |") {
	    warn "cannot list `$file': $!\n";
	    next;
	}
    }
    else {
	warn "no such file or directory: `$file'";
	next;
    }
    
    while (<LST>) {
	chomp;
	if (exists($db{"$_\0"})) {
	    ($opkg = $db{"$_\0"}) =~ s/\0//;
	    warn "overwriting `$_' (old: $opkg, new: $pkg)\n";
	}
	print "$pkg: $_\n" if ($verbose);
	$db{"$_\0"} = "$pkg\0";
    }

    close LST;
}

#
# cleanup
#

untie %db;
